canonical-web-and-design/maas-ui
# feat(machine): add filter groups actions

## Done

- Add support for fetching filter groups to the machine slice.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the redux dev tools and dispatch the following:
```
{
  type: 'machine/filterGroups',
  payload: null,
  meta: {
    model: 'machine',
    method: 'filter_groups'
  }
}
```
- Now look in the redux state for machines -> filters and you should see a list of all the groups.

## Fixes

Fixes: canonical-web-and-design/app-tribe#1140.
